test/browser.test.js

```diff
@@ -178,6 +178,51 @@ describe(`${browserString}, target: static, trailingSlash: true`, () => {
   })
 })
 
+describe(`${browserString}, target: static, trailingSlash: true and strategy prefix`, () => {
+  /** @type {Nuxt} */
+  let nuxt
+  /** @type {import('playwright-chromium').ChromiumBrowser} */
+  let browser
+
+  beforeAll(async () => {
+    const overrides = {
+      target: 'static',
+      router: {
+        trailingSlash: true
+      },
+      i18n: {
+        defaultLocale: 'en',
+        strategy: 'prefix'
+      }
+    }
+    nuxt = (await setup(loadConfig(__dirname, 'basic', overrides, { merge: true }))).nuxt
+    browser = await createBrowser()
+  })
+
+  afterAll(async () => {
+    if (browser) {
+      await browser.close()
+    }
+
+    await nuxt.close()
+  })
+
+  // Issue https://github.com/nuxt-community/i18n-module/issues/1204
+  test('redirects from root (404) path to default locale while preserving query', async () => {
+    const page = await browser.newPage()
+    await page.goto(url('/?key=value'))
+    expect(await (await page.$('body'))?.textContent()).toContain('locale: en')
+    expect(await getRouteFullPath(page)).toBe('/en/?key=value')
+  })
+
+  test('redirects from root (404) path to default locale while preserving query and hash', async () => {
+    const page = await browser.newPage()
+    await page.goto(url('/?key=value#hash'))
+    expect(await (await page.$('body'))?.textContent()).toContain('locale: en')
+    expect(await getRouteFullPath(page)).toBe('/en/?key=value#hash')
+  })
```

Comment:
Those tests pass even without the fix so probably not testing the right thing?

Reply:
Sorry, there is no problem with the static version, only on the server version.

I will update the tests ASAP.

Thanks!